core/src/main/java/hudson/model/Run.java

```diff
@@ -1787,6 +1789,13 @@ protected final void execute(@Nonnull RunExecution job) {
                     listener.finished(result);
                     listener.closeQuietly();
                 }
+                if (logger != null) {
```

Comment:
True, but that was already the case; this patch is merely improving the behavior when `listener.close()` does not call `logger.close()`.